code/game/machinery/doors/firedoor.dm

```diff
@@ -203,30 +201,67 @@
 		for(var/obj/machinery/firealarm/fire_panel in place.firealarms)
 			fire_panel.set_status()
 
-/obj/machinery/door/firedoor/proc/check_atmos(datum/source)
+/obj/machinery/door/firedoor/proc/refresh_shared_turfs(datum/source, list/leaving_members, list/joining_members)
+	SIGNAL_HANDLER
+	var/datum/merger/temp_group = source
+	if(temp_group.origin != src)
+		return
+	var/list/shared_problems = list() // We only want to do this once, this is a nice way of pulling that off
+	for(var/obj/machinery/door/firedoor/firelock as anything in merge_group.members)
+		firelock.issue_turfs = shared_problems
+
+/obj/machinery/door/firedoor/proc/check_adjacent_turfs()
+	for(var/dir in GLOB.cardinals)
+		var/turf/checked_turf = get_step(get_turf(src),dir)
+		if(checked_turf)
+			RegisterSignal(checked_turf, COMSIG_TURF_EXPOSE, .proc/process_results, override = TRUE)
+			RegisterSignal(checked_turf, COMSIG_TURF_CALCULATED_ADJACENT_ATMOS, .proc/process_results, override = TRUE)
+
+/obj/machinery/door/firedoor/proc/check_atmos(turf/checked_turf)
+	var/datum/gas_mixture/environment = checked_turf.return_air()
+	var/obj/machinery/airalarm/air_alarm = LAZYACCESS(my_area.airalarms,1)
+	
+	if(air_alarm)
+		var/datum/tlv/selected_temperature = air_alarm.TLV["temperature"]
+		if(environment?.temperature >= selected_temperature?.hazard_max)
+			return FIRELOCK_ALARM_TYPE_HOT
+		if(environment?.temperature <= selected_temperature?.hazard_min)
+			return FIRELOCK_ALARM_TYPE_COLD
+	else
+		if(environment?.temperature >= FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
+			return FIRELOCK_ALARM_TYPE_HOT
+		if(environment?.temperature <= BODYTEMP_COLD_DAMAGE_LIMIT)
```

Comment:
No, just remove them from the problem spots.

Reply:
cant I use this too?